pkg/client/factory_test.go

```diff
@@ -113,14 +115,23 @@ func TestClientWithDecisionServiceAndEventProcessorInOptions(t *testing.T) {
 
 func TestClientWithCustomCtx(t *testing.T) {
 	factory := OptimizelyFactory{}
-	testExecutionCtx := utils.NewCancelableExecutionCtx()
+	ctx, cancel := context.WithCancel(context.Background())
 	mockConfigManager := new(MockProjectConfigManager)
 	client, err := factory.Client(
 		WithConfigManager(mockConfigManager),
-		WithExecutionContext(testExecutionCtx),
+		WithContext(ctx),
 	)
 	assert.NoError(t, err)
-	assert.Equal(t, client.executionCtx, testExecutionCtx)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	client.execGroup.Go(func(ctx context.Context) {
+		<-ctx.Done()
+		wg.Done()
+	})
+
+	cancel()
```

Comment:
cool, we were missing that in the ExecutionContext implementation. The only awkward thing will happen when a user will define cancelable context and cancels with the client: 

```
ctx, _ := context.WithCancel(context.Background())
client.Close()
```
that will return :

 ```the cancel function returned by context.WithCancel should be called, not discarded, to avoid a context leak```